dvc/repo/plots/__init__.py

```diff
@@ -53,18 +53,31 @@ def collect(self, targets=None, revs=None):
             tree = RepoTree(self.repo)
             plots = _collect_plots(self.repo, targets, rev)
             for path_info, props in plots.items():
-                datafile = relpath(path_info, self.repo.root_dir)
+
                 if rev not in data:
                     data[rev] = {}
-                data[rev].update({datafile: {"props": props}})
 
-                # Load data from git or dvc cache
-                try:
-                    with tree.open(path_info) as fd:
-                        data[rev][datafile]["data"] = fd.read()
-                except FileNotFoundError:
-                    # This might happen simply because cache is absent
-                    pass
+                if tree.isdir(path_info):
+                    plot_files = []
+                    for pi in tree.walk_files(path_info):
+                        plot_files.append(
+                            (pi, relpath(pi, self.repo.root_dir))
+                        )
+                else:
+                    plot_files = [
+                        (path_info, relpath(path_info, self.repo.root_dir))
+                    ]
+
+                for path, repo_path in plot_files:
+                    data[rev].update({repo_path: {"props": props}})
+
+                    # Load data from git or dvc cache
+                    try:
+                        with tree.open(path) as fd:
+                            data[rev][repo_path]["data"] = fd.read()
+                    except FileNotFoundError:
+                        # This might happen simply because cache is absent
+                        pass
```

Comment:
We start to support `dir` plots: all the files in the directory marked as `plots` will be treated the same way, and additional config provided with `dvc plots modify` will be applied to each one of them.
I intend to write support for "granular" config in such use case, though it will probably require some reworking on `dvc.yaml` side and is not essential to make logs/live metrics work.